rts/Lua/LuaMatrix.hpp

```diff
@@ -0,0 +1,197 @@
+	}
+	LuaMatrixImpl Mult(const LuaMatrixImpl& mat2) { return mat * mat2.GetMatRef(); }
+
+	LuaMatrixImpl Add(const LuaMatrixImpl& mat2) { return mat + mat2.GetMatRef(); }
+	*/
+
+	void InverseAffine() { mat.InvertAffineInPlace(); };
+	void Inverse() { mat.InvertInPlace(); };
+
+	void Transpose() { mat.Transpose(); };
+
+	void Translate(const float x, const float y, const float z);
+	void Scale(const float x, const float y, const float z);
+	void Scale(const float s);
+
+	void RotateRad(const float rad, const float x, const float y, const float z);
+	void RotateDeg(const float deg, const float x, const float y, const float z) { RotateRad(deg * math::DEG_TO_RAD, x, y, z); };
+
+	void RotateRadX(const float rad) { RotateRad(rad, 1.0f, 0.0f, 0.0f); };
+	void RotateRadY(const float rad) { RotateRad(rad, 0.0f, 1.0f, 0.0f); };
+	void RotateRadZ(const float rad) { RotateRad(rad, 0.0f, 0.0f, 1.0f); };
+
+	void RotateDegX(const float deg) { RotateRad(deg * math::DEG_TO_RAD, 1.0f, 0.0f, 0.0f); };
+	void RotateDegY(const float deg) { RotateRad(deg * math::DEG_TO_RAD, 0.0f, 1.0f, 0.0f); };
+	void RotateDegZ(const float deg) { RotateRad(deg * math::DEG_TO_RAD, 0.0f, 0.0f, 1.0f); };
+
+	bool UnitMatrix(const unsigned int unitID, const sol::optional<bool> mult, sol::this_state L) { return ObjectMatImpl<CUnit>(unitID, mult.value_or(objectMultDefault), L); }
+	bool UnitPieceMatrix(const unsigned int unitID, const unsigned int pieceNum, const sol::optional<bool> mult, sol::this_state L) { return ObjectPieceMatImpl<CUnit>(unitID, pieceNum, mult.value_or(objectMultDefault), L); }
+
+	bool FeatureMatrix(const unsigned int featureID, const sol::optional<bool> mult, sol::this_state L) { return ObjectMatImpl<CFeature>(featureID, mult.value_or(objectMultDefault), L); }
+	bool FeaturePieceMatrix(const unsigned int featureID, const unsigned int pieceNum, const sol::optional<bool> mult, sol::this_state L) { return ObjectPieceMatImpl<CFeature>(featureID, pieceNum, mult.value_or(objectMultDefault), L); }
+
+	bool ProjectileMatrix(const unsigned int projID, const sol::optional<bool> mult, sol::this_state L) { return ObjectMatImpl<CFeature>(projID, mult.value_or(objectMultDefault), L); }
+	bool ProjectilePieceMatrix(const unsigned int projID, const unsigned int pieceNum, const sol::optional<bool> mult, sol::this_state L) { return ObjectPieceMatImpl<CFeature>(projID, pieceNum, mult.value_or(objectMultDefault), L); } //does this exist at all?
+
+	void ScreenViewMatrix(const sol::optional<bool> mult);
+	void ScreenProjMatrix(const sol::optional<bool> mult);
+
+	void SunViewMatrix(const sol::optional<bool> mult);
+	void SunProjMatrix(const sol::optional<bool> mult);
+
+	void Ortho(const float left, const float right, const float bottom, const float top, const float near, const float far, const sol::optional<bool> mult);
+	void Frustum(const float left, const float right, const float bottom, const float top, const float near, const float far, const sol::optional<bool> mult);
```

Comment:
In function declaration, don't put const on parameters that are passed by value as it has no meaning: pass-by-value is always going to be a copy so the caller cannot distinguish between const and non-const pass-by-value parameters. 

It's perfectly fine to do this in function *definitions* as that describes how a value is used within the function (it mainly helps the reader understand the function body)